tico/passes/remove_redundant_permute.py

```diff
@@ -61,39 +81,37 @@ def call(self, exported_program: ExportedProgram) -> PassResult:
                 continue
             if len(permute2.users) != 1:
                 continue
-            assert len(permute2.args) == 2
-            permute1, permute2_dims = permute2.args
-            assert isinstance(permute1, torch.fx.Node), type(permute1)
-            assert isinstance(permute2_dims, list), type(permute2_dims)
-            for dim in permute2_dims:
-                assert isinstance(dim, int), type(dim)
+            permute_args = PermuteArgs(*permute2.args, **permute2.kwargs)  # type: ignore[arg-type]
+            permute1, permute2_dims = permute_args.input, permute_args.dims
 
             if not permute1.target in ops.aten.permute:
                 continue
             if len(permute1.users) != 1:
                 continue
-            assert len(permute1.args) == 2
-            permute1_input, permute1_dims = permute1.args
-            assert isinstance(permute1_input, torch.fx.Node), type(permute1_input)
-            assert isinstance(permute1_dims, list), type(permute1_dims)
-            for dim in permute1_dims:
-                assert isinstance(dim, int), type(dim)
-
-            # shape
-            permute1_input_shape = extract_shape(permute1_input)
-            permute2_shape = extract_shape(permute2)
-            if permute1_input_shape != permute2_shape:
-                continue
-            # stride
-            permute1_input_stride = extract_stride(permute1_input)
-            permute2_stride = extract_stride(permute2)
-            if permute1_input_stride != permute2_stride:
-                continue
-
-            permute2.replace_all_uses_with(permute1_input, propagate_meta=False)
-
+            permute1_args = PermuteArgs(*permute1.args, **permute1.kwargs)  # type: ignore[arg-type]
+            permute1_input, permute1_dims = permute1_args.input, permute1_args.dims
+
+            fused_dims = _compose_permutation(permute1_dims, permute2_dims)
+            identity = list(range(len(fused_dims)))
+
+            if fused_dims == identity:
+                # shape
+                permute1_input_shape = extract_shape(permute1_input)
+                permute2_shape = extract_shape(permute2)
+                if permute1_input_shape != permute2_shape:
+                    continue
```

Comment:
Is this case possible?

Reply:
Ah, it is from old condition. Seems that it can be just an assertion.